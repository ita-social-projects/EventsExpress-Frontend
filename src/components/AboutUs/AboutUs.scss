@import "../../styles/variables";

.about__container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 50px;
}

.about__info {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-basis: 50%;
}

.about__info_text {
  width: 515px;
  font-weight: 600;
  margin-bottom: 15px;
}

.about__info__button {
  display: flex;
  justify-content: flex-end;
  align-self: flex-end;
  font-weight: bold;
  font-size: 14px;
  color: $lightThemeAccept;
  border: 2px solid $lightThemeAccept;
  border-radius: $buttonBorderRadius;
  padding: 5px 25px;
  cursor: pointer;
  transition: 0.5s all;

  &:hover {
    color: darken($color: $lightThemeAccept, $amount: 15%);
    border-color: darken($color: $lightThemeAccept, $amount: 15%);
    background-color: #e6e6e6;
    text-decoration: none;
  }
}

.about__header {
  position: relative;
  display: flex;
  color: $black;
  font-weight: bold;
  font-size: 72px;
  margin: 0;
  padding: 0;
  margin-bottom: 20px;
  user-select: none;

  &::before {
    content: "";
    position: absolute;
    left: 65px;
    top: 3px;
    height: 75px;
    width: 150px;
    background-color: $lightThemeAccept;
    z-index: -1;
  }
}

.about__photos {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
}

.about__photos_item {
  position: relative;
  overflow: hidden;
  width: 30%;
  height: 150px;
  margin: 0 10px;
  margin-bottom: 20px;
}

.about__photos_text {
  position: absolute;
  font-weight: bold;
  font-size: 24px;
  bottom: 7px;
  right: 0px;
  color: $white;
  background: $transparentBlackBackground;
  padding: 5px;
  width: 150px;
  user-select: none;
}
