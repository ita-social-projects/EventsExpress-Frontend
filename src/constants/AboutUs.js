const ABOUT_US = {
  JOIN_US: "Join us",
  ABOUT_US: "About us",
  CONTACT_US: "Contact us",
  SEARCH_EVENTS: "Search events",
  CREATE_YOUR_EVENT: "Create your event",
  FIND_EVENTS: "Find events",
  GET_CONNECTED: "Get connected",
  HAVE_FUN_TOGETHER: "Have fun together",
  LOREM_IPSUM:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim adminim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt inculpa qui officia deserunt mollit anim id est laborum. ",
};

export default ABOUT_US;
