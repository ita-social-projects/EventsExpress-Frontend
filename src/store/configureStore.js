import { applyMiddleware, combineReducers, compose, createStore } from "redux";
import thunk from "redux-thunk";
import { routerMiddleware } from "react-router-redux";
import rootReducers from "../reducers/root";

export default function configureStore(history, initialState) {
  const middleware = [thunk, routerMiddleware(history)];

  // In development, use the browser's Redux dev tools extension if installed
  const isDevelopment = process.env.NODE_ENV === "development";
  const composeEnhancers =
    isDevelopment &&
    typeof window !== "undefined" &&
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose);
  const rootReducer = combineReducers({
    ...rootReducers,
  });

  return createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(...middleware)),
  );
}
