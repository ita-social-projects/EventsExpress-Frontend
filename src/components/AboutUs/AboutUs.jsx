import React from "react";
import "./AboutUs.scss";
import aboutImg1 from "./images/aboutImg-1.png";
import aboutImg2 from "./images/aboutImg-2.png";
import aboutImg3 from "./images/aboutImg-3.png";
import aboutImg4 from "./images/aboutImg-4.png";
import aboutImg5 from "./images/aboutImg-5.png";
import constants from "../../constants/AboutUs";

const {
  ABOUT_US,
  SEARCH_EVENTS,
  CREATE_YOUR_EVENT,
  FIND_EVENTS,
  GET_CONNECTED,
  HAVE_FUN_TOGETHER,
} = constants;

const AboutUs = () => {
  const MAPPER = [
    { img: aboutImg1, text: SEARCH_EVENTS },
    { img: aboutImg2, text: CREATE_YOUR_EVENT },
    { img: aboutImg3, text: FIND_EVENTS },
    { img: aboutImg4, text: GET_CONNECTED },
    { img: aboutImg5, text: HAVE_FUN_TOGETHER },
  ];

  return (
    <div className="container about__container">
      <div className="about__info">
        <h1 className="about__header"> {ABOUT_US} </h1>
        <div className="about__info_text">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
          aliquip ex ea commodo consequat. Duis aute irure dolor in
          reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
          culpa qui officia deserunt mollit anim id est laborum.
        </div>
        <button
          className="about__info__button"
          type="button"
          onClick={() => {
            console.log(123);
          }}
        >
          Join us
        </button>
      </div>
      <div className="about__photos">
        {MAPPER.map(({ img, text }) => (
          <div key={text} className="about__photos_item">
            <img src={img} alt={text} />
            <div className="about__photos_text">{text}</div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default AboutUs;
