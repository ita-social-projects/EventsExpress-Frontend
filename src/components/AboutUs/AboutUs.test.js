/* eslint-disable prefer-const */
import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import AboutUs from "./AboutUs";

const mockStore = configureStore([]);

it("renders when user don`t registered", () => {
  let store;
  store = mockStore({
    user: {},
  });
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <AboutUs />
        </Provider>
        ,
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders when user registered", () => {
  let store;
  store = mockStore({
    user: {
      id: 1,
      name: "Anton",
    },
  });
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <AboutUs user={store.user} />
        </Provider>
        ,
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
