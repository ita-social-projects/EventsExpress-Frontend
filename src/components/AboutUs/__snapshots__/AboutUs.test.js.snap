// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AboutUs component renders when user didn\`t register 1`] = `
Array [
  <div
    className="container about__container"
  >
    <div
      className="about__info"
    >
      <h1
        className="about__header"
      >
         
        About us
         
      </h1>
      <div
        className="about__info_text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </div>
      <button
        className="about__info__button"
        onClick={[Function]}
        type="button"
      >
        Join us
      </button>
    </div>
    <div
      className="about__photos"
    >
      <div
        className="about__photos_item"
      >
        <img
          alt="Search events"
          src="aboutImg-1.png"
        />
        <div
          className="about__photos_text"
        >
          Search events
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Create your event"
          src="aboutImg-2.png"
        />
        <div
          className="about__photos_text"
        >
          Create your event
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Find events"
          src="aboutImg-3.png"
        />
        <div
          className="about__photos_text"
        >
          Find events
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Get connected"
          src="aboutImg-4.png"
        />
        <div
          className="about__photos_text"
        >
          Get connected
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Have fun together"
          src="aboutImg-5.png"
        />
        <div
          className="about__photos_text"
        >
          Have fun together
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;

exports[`Render AboutUs component renders when user registered 1`] = `
Array [
  <div
    className="container about__container"
  >
    <div
      className="about__info"
    >
      <h1
        className="about__header"
      >
         
        About us
         
      </h1>
      <div
        className="about__info_text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </div>
      <a
        aria-current={null}
        className="about__info__button"
        href="/contactAdmin"
        onClick={[Function]}
      >
        Contact us
      </a>
    </div>
    <div
      className="about__photos"
    >
      <div
        className="about__photos_item"
      >
        <img
          alt="Search events"
          src="aboutImg-1.png"
        />
        <div
          className="about__photos_text"
        >
          Search events
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Create your event"
          src="aboutImg-2.png"
        />
        <div
          className="about__photos_text"
        >
          Create your event
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Find events"
          src="aboutImg-3.png"
        />
        <div
          className="about__photos_text"
        >
          Find events
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Get connected"
          src="aboutImg-4.png"
        />
        <div
          className="about__photos_text"
        >
          Get connected
        </div>
      </div>
      <div
        className="about__photos_item"
      >
        <img
          alt="Have fun together"
          src="aboutImg-5.png"
        />
        <div
          className="about__photos_text"
        >
          Have fun together
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;
